From: Jan Kara <jack@suse.com>
Patch-mainline: Never, fixed differently by 5accdf82ba25a and company
References: bsc#935123
Subject: writeback: Skip writeback for frozen filesystem

Skip background writeback (writeback using pin_sb_for_writeback()) for
frozen filesystem so that we don't block on frozen filesystem while holding
s_umount semaphore.

Signed-off-by: Jan Kara <jack@suse.com>

---
 fs/fs-writeback.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/fs-writeback.c
+++ b/fs/fs-writeback.c
@@ -504,7 +504,7 @@ static bool pin_sb_for_writeback(struct
 	spin_unlock(&sb_lock);
 
 	if (down_read_trylock(&sb->s_umount)) {
-		if (sb->s_root)
+		if (sb->s_frozen == SB_UNFROZEN && sb->s_root)
 			return true;
 		up_read(&sb->s_umount);
 	}
